kyo-prelude/shared/src/main/scala/kyo/Guard.scala

```diff
@@ -0,0 +1,68 @@
+package kyo
+
+import kyo.Guard.CanGuard
+import kyo.kernel.ArrowEffect
+
+/** Protect effect(s) [[S]] from being handled. To be used in functions that accept effects with generic effect types in order to avoid
+  * accidentally handling them. Can only be used within [[Guard.use]] blocks.
+  *
+  * @tparam S
+  *   The effect type that is guarded
+  * @see
+  *   [[Guard]]
+  */
+sealed trait Guarded[S] extends ArrowEffect[[A] =>> A < S, Id]
+
+/** Utility for protecting any effect of type [[S]] from being handled
+  *
+  * @tparam S
+  *   The effect type that can be guarded
+  */
+sealed class Guard[S](using tag: Tag[Guarded[S]]):
```

Comment:
could we avoid allocating this class? I think something like this could work:

```scala
opaque type Guard[S] = Unit

object Guard:
  extension [S](self: Guard[S])
     def apply[A](effect: A < S)(using CanGuard, Frame, Tag[Guard[S]]): A < Guarded[S]
```